File: challenges/linear-search/js/linearSearchSpec.js
var ls = require("./linearSearch");
var shallowEqualArrays = require('shallow-equal/arrays');

console.log(`${ls.linearSearch(3, [1, 2, 3]) === 2}: ${ls.linearSearch(3, [1, 2, 3])}`);
console.log(`${ls.linearSearch(4, [1, 2, 3]) === undefined}: ${ls.linearSearch(4, [1, 2, 3])}`);
console.log(`${ls.linearSearch(13, [1, 2, 3]) === undefined}: ${ls.linearSearch(13, [1, 2, 3])}`);
console.log(`${ls.linearSearch(1, [1, 2, 3]) === 0}: ${ls.linearSearch(1, [1, 2, 3])}`);
console.log(`${ls.linearSearch(4, [1, 2, 3, 4]) === 3}: ${ls.linearSearch(4, [1, 2, 3, 4])}`);

// console.log(shallowEqualArrays(ls.linearSearchGlobal("a", ["b", "a", "n", "a", "n", "a", "s"]), [1, 3, 5]))
// console.log(shallowEqualArrays(ls.linearSearchGlobal("s", ["b", "a", "n", "a", "n", "a", "s"]), [6]))
// console.log(shallowEqualArrays(ls.linearSearchGlobal("n", ["b", "a", "n", "a", "n", "a", "s"]), [2, 4]))

File: challenges/linear-search/python/linear_search_spec.py
from linear_search import linear_search, global_linear_search

print(linear_search(3, [1, 2, 3]) == 2)
print(linear_search(4, [1, 2, 3]) == None)
print(linear_search(13, [1, 2, 3]) == None)

print(global_linear_search("a", 'bananas') == [1, 3, 5])
print(global_linear_search(3, [1, 2, 3, 4, 5, 3]) == [2, 5])
print(global_linear_search(0, [1, 2, 3, 4, 5, 3]) == [])
print(global_linear_search(1, []) == [])
